pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4209 @@
+            CIMError: CIM_ERR_INVALID_PARAMETER
+        """
+
+        self.validate_namespace(namespace)
+        if isinstance(ObjectName, CIMInstanceName):
+            self._validate_instancename_namespace(namespace, ObjectName)
+            rtn_paths = self._get_associated_instancenames(namespace,
+                                                           ObjectName,
+                                                           AssocClass,
+                                                           ResultClass,
+                                                           ResultRole, Role)
+            results = [p.copy() for p in rtn_paths]
+
+            for iname in results:
+                if iname.host is None:
+                    iname.host = self.host
+
+        else:
+            rtn_classnames = self._get_associated_classnames(namespace,
+                                                             ObjectName,
+                                                             AssocClass,
+                                                             ResultClass,
+                                                             ResultRole, Role)
+            # TODO: Future Returns CIMClassName. Should rtn just the classname
+            # from a server since CIMClassName is a client artifact and the
+            # server has no way to send that to client.
+            results = [CIMClassName(classname=cln, host=self.host,
+                                    namespace=namespace)
+                       for cln in rtn_classnames]
+
+        return results
+
+    ####################################################################
+
+    #   Associators server request
+    #
+    ####################################################################
+
+    def Associators(self, namespace, ObjectName, AssocClass=None,
+                    ResultClass=None,
+                    Role=None, ResultRole=None, IncludeQualifiers=None,
+                    IncludeClassOrigin=None, PropertyList=None):
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.Associators`.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing slash
+            characters are ignored.
+
+          ObjectName:
+            The object path of the source object:
+
+            * Instance-level use: The instance path of the
+              source instance, as a :class:`~pywbem.CIMInstanceName` object. If
+              this object specifies a namespace, it must match the namespace
+              parameter. Its `host` attribute will be ignored.
+
+            * Class-level use: The class path of the source
+              class, as a :term:`string`. The string is interpreted as a class
+              name in the default namespace of the connection
```

Comment:
On "the default namespace of the connection": Change to "the namespace specified in the namespace parameter".

Reply:
DONE